### ISSUE TITLE
Flaky test TestNoSA1503IfStatementsAsync

### ISSUE THREAD

**Issue by tmaczynski** [DotNetAnalyzers/StyleCopAnalyzers#2129]
I'm working on issue #1562 and sometimes I ran all unit tests to check if I've broken something. From time to time, some test randomly fails. Here's an example of a message from a failed test:

```
Test Name:	StyleCop.Analyzers.Test.LayoutRules.SA1501UnitTests.TestNoSA1503IfStatementsAsync
Test FullName:	StyleCop.Analyzers.Test.LayoutRules.SA1501UnitTests.TestNoSA1503IfStatementsAsync
Test Source:	C:\Users\Tomek\programowanie\dot_net_analysers\StyleCopAnalyzers\StyleCop.Analyzers\StyleCop.Analyzers.Test\LayoutRules\SA1501UnitTests.cs : line 619
Test Outcome:	Failed
Test Duration:	0:00:00.12

Result StackTrace:	
w Xunit.Assert.Equal(String expected, String actual, Boolean ignoreCase, Boolean ignoreLineEndingDifferences, Boolean ignoreWhiteSpaceDifferences) w C:\BuildAgent\work\cb37e9acf085d108\src\xunit.assert\Asserts\StringAsserts.cs:wiersz 239
   w Xunit.Assert.Equal(String expected, String actual) w C:\BuildAgent\work\cb37e9acf085d108\src\xunit.assert\Asserts\StringAsserts.cs:wiersz 169
   w TestHelper.CodeFixVerifier.<VerifyFixInternalAsync>d__14.MoveNext() w C:\Users\Tomek\programowanie\dot_net_analysers\StyleCopAnalyzers\StyleCop.Analyzers\StyleCop.Analyzers.Test\Verifiers\CodeFixVerifier.cs:wiersz 355
--- Koniec śladu stosu z poprzedniej lokalizacji, w której wystąpił wyjątek ---
   w System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   w System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   w System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()
   w TestHelper.CodeFixVerifier.<VerifyCSharpFixAsync>d__5.MoveNext() w C:\Users\Tomek\programowanie\dot_net_analysers\StyleCopAnalyzers\StyleCop.Analyzers\StyleCop.Analyzers.Test\Verifiers\CodeFixVerifier.cs:wiersz 89
--- Koniec śladu stosu z poprzedniej lokalizacji, w której wystąpił wyjątek ---
   w System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   w System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   w System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()
   w StyleCop.Analyzers.Test.LayoutRules.SA1501UnitTests.<TestNoSA1503IfStatementsAsync>d__23.MoveNext() w C:\Users\Tomek\programowanie\dot_net_analysers\StyleCopAnalyzers\StyleCop.Analyzers\StyleCop.Analyzers.Test\LayoutRules\SA1501UnitTests.cs:wiersz 720
--- Koniec śladu stosu z poprzedniej lokalizacji, w której wystąpił wyjątek ---
   w System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   w System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   w Xunit.Sdk.TestInvoker`1.<>c__DisplayClass46_1.<<InvokeTestMethodAsync>b__1>d.MoveNext()
--- Koniec śladu stosu z poprzedniej lokalizacji, w której wystąpił wyjątek ---
   w System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   w System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   w Xunit.Sdk.ExecutionTimer.<AggregateAsync>d__4.MoveNext()
--- Koniec śladu stosu z poprzedniej lokalizacji, w której wystąpił wyjątek ---
   w System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   w System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   w Xunit.Sdk.ExceptionAggregator.<RunAsync>d__9.MoveNext() w C:\BuildAgent\work\cb37e9acf085d108\src\xunit.core\Sdk\ExceptionAggregator.cs:wiersz 90
Result Message:	
Assert.Equal() Failure
                                   ↓ (pos 913)
Expected: ···          else\r\n                Debug.Assert(false);\r\n       ···
Actual:   ···          else\r\n    Debug.Assert(false);\r\nelse\r\n            D···
                                   ↑ (pos 913)
```

This message is localized, here's an English translation of relevant parts:
_Koniec śladu stosu z poprzedniej lokalizacji, w której wystąpił wyjątek_ means _End of stack trace from previous location where exception was thrown_
_w_ means _in_
_wiersz_ means _line_

Here's link to [a commit at my fork](https://github.com/tmaczynski/StyleCopAnalyzers/commit/68b9a200f6dc8845bf6610c9a94f7a12f9eea2d7) where the test failed.

I'm using Test Explorer in Visual Studio update 2 with Resharper 2016.1.1 on Windows 10 to run tests. I only observed this behavior when I perform "Run all" and I cannot reproduce a failure (but it might be due to a low probability of a test failure).

**Comment by vweijsters** [comment]
Thanks for reporting this. I'll mark it as a bug.

**Comment by vweijsters** [comment]
Further analysis shows that this actually is a duplicate of #2010 and it will be fixed when #2067 is merged.
